cli/src/pcluster/config/cluster_config.py

```diff
@@ -1181,6 +1210,7 @@ def __init__(
         self.compute_startup_time_metric_enabled = Resource.init_param(
             compute_startup_time_metric_enabled, default=False
         )
+        self.munge_key_settings = munge_key_settings or SlurmSettingsForCustomMungeKey(implied=True)
```

Comment:
Are you sure we need the `or SlurmSettingsForCustomMungeKey(implied=True)` bit? What does it do exactly? Doesn't it create a new object if the `munge_key_settings` object is missing? But if it's missing, it means that the custom munge key part of the config is not necessary, right? Or am I missing something here?

Reply:
> What does it do exactly?

```
  munge_key_manager 'manage_munge_key' do
    munge_key_secret_arn lazy {
      node['cluster']['config'].dig(:DevSettings, :MungeKeySettings, :MungeKeySecretArn)
    }
  end
```
In the cookbook recipe, we will need `MungeKeySecretArn` as a input in some logic. eg: setup munge key and the rotation script, check if `MungeKeySecretArn` update funciton. 
So `SlurmSettingsForCustomMungeKey(implied=True)` Can ensure at least we have a
```
DevSettings:
  MungeKeySettings:
    MungeKeySecretArn: null
```
in the `cluster-config.yaml` in the shared directory.
Or it will not contain `MungeKeySettings` section. The `dig` method will raise `no method dig for nil class` error. If I don't remember wrong.

Reply:
Exactly, but we can still deal with that logic by checking if the parent section of the dictionary exists.

Still, let's stick to what you implemented right now.